### PR TITLE
Create documentation page with glossary of terms

### LINKED ISSUE #42: Create documentation page with glossary of terms
<a href="https://github.com/Mr-Geekman"></img></a> **Issue by [Mr-Geekman](https://github.com/Mr-Geekman)**
_Monday Aug 07, 2023 at 08:49 GMT_
_Originally opened as https://github.com/tinkoff-ai/etna/issues/1349_

----

### 🚀 Feature Request

We have some kind of glossary in the library, but it isn't explicit. We could make it explicit as a separate documentation page.

### Proposal

Add page "User Guide / Glossary" with explanation of terms:
- Time series
- Frequency
- Univariate time series
- Panel time series
- Multivariate time series
- Hierarchical time series
- Segments
- Endogenous data
- Exogenous data
- Regressors
- Stationarity
- Seasonality
- Trend
- Anomaly (?)
- Change-point
- Prediction intervals
- Prediction components
- Forecasting
- Horizon
- Backtesting
- Per-segment / local mode
- Multi-segment / Global mode
- Forecasting strategy
- Clustering
- Classification
- Segmentation
- Dataset (?)
- Transform (?)
- Model (?)
- Model context (?)
- Pipeline (?)
- Lags (?)
- Date flags (?)
- Fourier terms (?)
- Differencing (?)

Inspiration: https://www.sktime.net/en/stable/glossary.html

To discuss:
- Do we want to add terms marked with (?) ?
- Do we want to list them in some particular order or in alphabetical order?
- Do we want to add some other terms?

### Test cases

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
See #42.

## Closing issues
Closes #42.

### CODE REVIEW

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+   Lags
+      The features generated by :py:class:`~etna.transforms.math.lags.LagTransform`.
+
+   Date flags
+      The features generated by :py:class:`~etna.transforms.timestamp.date_flags.DateFlagsTransform`.
+
+   Fourier terms
+      The features generated by :py:class:`~etna.transforms.timestamp.fourier.FourierTransform`.
+
+   Differencing
```
Comment (brsnw250) [inline docs/source/glossary.rst:129]: Maybe here higlight seasonal differencing (`period > 1`)?

File: docs/source/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      High-level entity for solving :term:`forecasting` task. Works with :term:`dataset`, :term:`model`, :term:`transforms <transform>` and other :term:`pipelines <pipeline>`.
+
+   Lags
+      The features generated by :py:class:`~etna.transforms.math.lags.LagTransform`.
+
+   Date flags
+      The features generated by :py:class:`~etna.transforms.timestamp.date_flags.DateFlagsTransform`.
+
+   Fourier terms
+      The features generated by :py:class:`~etna.transforms.timestamp.fourier.FourierTransform`.
```
Comment (brsnw250) [inline docs/source/glossary.rst:127]: Should we add a bit more detailed definition? Right now it is not very clear what these values are.

**Review by brsnw250 — CHANGES_REQUESTED** [review]

File: docs/source/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      It doesn't have to be always the same number of seconds.
+      For example, taking the first day of each month is a valid frequency.
+
+   Univariate time series
+      A single :term:`time series` containing measurements of a scalar variable.
+
+   Multivariate time series
+      A single :term:`time series` containing measurements of a multidimensional variable.
+
+   Panel time series
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply docs/source/glossary.rst:24]: I think this point is good for educational purposes. People make this distinction and we have been questioned about this in the chat.

File: docs/source/glossary.rst (hunk 2 above)
Comment (Mr-Geekman) [reply docs/source/glossary.rst:127]: Users could read it in the documentation of `FourierTransform` I suppose.

**Review by brsnw250 — APPROVED** [review]

**Pull request merged**